spring-data-jdbc/src/test/java/org/springframework/data/jdbc/repository/support/JdbcQueryMethodUnitTests.java

```diff
@@ -57,15 +65,92 @@ public void returnsSpecifiedRowMapperClass() throws NoSuchMethodException {
+		Properties properties = new Properties();
+		properties.setProperty(DUMMY_SELECT_NAME, DUMMY_SELECT_VALUE);
+		NamedQueries nameQueries = new PropertiesBasedNamedQueries(properties);
 
-		JdbcQueryMethod queryMethod = new JdbcQueryMethod(JdbcQueryMethodUnitTests.class.getDeclaredMethod("queryMethod"),
-				metadata, mock(ProjectionFactory.class));
+		JdbcQueryMethod queryMethod = new JdbcQueryMethod(
+				JdbcQueryMethodUnitTests.class.getDeclaredMethod("queryMethod"), metadata,
+				mock(ProjectionFactory.class), nameQueries);
 
 		assertThat(queryMethod.getRowMapperClass()).isEqualTo(CustomRowMapper.class);
 	}
 
-	@Query(value = DUMMY_SELECT, rowMapperClass = CustomRowMapper.class)
-	private void queryMethod() {}
+
+
+
+
+
+
+
+
+	@Test // DATAJDBC-234
+	public void returnsSqlStatementName() throws NoSuchMethodException {
+
+		RepositoryMetadata metadata = mock(RepositoryMetadata.class);
+
+		doReturn(String.class).when(metadata).getReturnedDomainClass(any(Method.class));
+
+		Properties properties = new Properties();
+		properties.setProperty(DUMMY_SELECT_NAME, DUMMY_SELECT_VALUE);
+		NamedQueries nameQueries = new PropertiesBasedNamedQueries(properties);
+
+		JdbcQueryMethod queryMethod = new JdbcQueryMethod(
+				JdbcQueryMethodUnitTests.class.getDeclaredMethod("queryMethodName"), metadata,
+				mock(ProjectionFactory.class), nameQueries);
+		System.out.println("getAnnotatedQuery: "+queryMethod.getAnnotatedQuery());
+		assertThat(queryMethod.getAnnotatedQuery()).isEqualTo(DUMMY_SELECT_VALUE);
+
+	}
+	@Test // DATAJDBC-234
+	public void returnsSqlStatementNameAndValue() throws NoSuchMethodException {
+
+		RepositoryMetadata metadata = mock(RepositoryMetadata.class);
+
+		doReturn(String.class).when(metadata).getReturnedDomainClass(any(Method.class));
+
+		Properties properties = new Properties();
+		properties.setProperty(DUMMY_SELECT_NAME, DUMMY_SELECT_VALUE);
+		NamedQueries nameQueries = new PropertiesBasedNamedQueries(properties);
+
+		JdbcQueryMethod queryMethod = new JdbcQueryMethod(
+				JdbcQueryMethodUnitTests.class.getDeclaredMethod("queryMethodNameAndValue"), metadata,
+				mock(ProjectionFactory.class), nameQueries);
+		assertThat(queryMethod.getAnnotatedQuery()).isEqualTo(DUMMY_SELECT_NAME_VALUE);
+
+	}
+
+	@Test // DATAJDBC-234
+	public void returnsNullNoSqlQuery() throws NoSuchMethodException {
+
+		RepositoryMetadata metadata = mock(RepositoryMetadata.class);
+		Properties properties = new Properties();
+		properties.setProperty(DUMMY_SELECT_NAME, DUMMY_SELECT_VALUE);
+		NamedQueries nameQueries = new PropertiesBasedNamedQueries(properties);
+
+		doReturn(String.class).when(metadata).getReturnedDomainClass(any(Method.class));
+
+		JdbcQueryMethod queryMethod = new JdbcQueryMethod(
+				JdbcQueryMethodUnitTests.class.getDeclaredMethod("queryWhitoutQueryAnnotation"), metadata,
+				mock(ProjectionFactory.class), nameQueries);
+		assertThat(queryMethod.getAnnotatedQuery()).isEqualTo("queryWhitoutQueryAnnotation");
+
+	}
+
```

Comment:
All these tests seem to miss the case where the named query to look up is derived directly from the method name.